README.md

```diff
@@ -44,30 +56,78 @@ Source Link in your IDE settings. Then you can debug GraphQL.NET source code as
 
 ## Installation
 
-You can install the latest stable version via [NuGet](https://www.nuget.org/packages/GraphQL/).
+#### 1. GraphQL.NET engine
+
+This is the main package, the heart of the repository in which you can find all the necessary classes
+for GraphQL request processing.
+
 ```
 > dotnet add package GraphQL
 ```
 
-For serialized results, you'll need an `IDocumentWriter` implementation.
-We support several serializers (or you can bring your own):
+#### 2. Serialization
 
-| Package | Downloads | Nuget Latest |
-|---------|-----------|--------------|
-| GraphQL.SystemTextJson | [![Nuget](https://img.shields.io/nuget/dt/GraphQL.SystemTextJson)](https://www.nuget.org/packages/GraphQL.SystemTextJson) | [![Nuget](https://img.shields.io/nuget/vpre/GraphQL.SystemTextJson)](https://www.nuget.org/packages/GraphQL.SystemTextJson) |
-| GraphQL.NewtonsoftJson | [![Nuget](https://img.shields.io/nuget/dt/GraphQL.NewtonsoftJson)](https://www.nuget.org/packages/GraphQL.NewtonsoftJson) | [![Nuget](https://img.shields.io/nuget/vpre/GraphQL.NewtonsoftJson)](https://www.nuget.org/packages/GraphQL.NewtonsoftJson) |
+For serialized results, you'll need an `IDocumentWriter` implementation.
+We provide several serializers (or you can bring your own).
 
 ```
 > dotnet add package GraphQL.SystemTextJson
 > dotnet add package GraphQL.NewtonsoftJson
 ```
+
 > *Note: You can use `GraphQL.NewtonsoftJson` with .NET Core 3+, just be aware it lacks async writing 
 > capabilities so writing to an ASP.NET Core 3.0 `HttpResponse.Body` will require you to set 
 > `AllowSynchronousIO` to `true` as per [this announcement](https://github.com/aspnet/Announcements/issues/342);
 > which isn't recommended.*
 
-You can get all preview versions from [GitHub Packages](https://github.com/orgs/graphql-dotnet/packages?repo_name=graphql-dotnet).
-Note that GitHub requires authentication to consume the feed.  See [here](https://docs.github.com/en/free-pro-team@latest/packages/publishing-and-managing-packages/about-github-packages#authenticating-to-github-packages).
+#### 3. Document Caching
+
+For caching of parsed GraphQL documents you'll need an `IDocumentCache` implementation.
+We provide in-memory implementation on top of `Microsoft.Extensions.Caching.Memory` package.
+
+```
+> dotnet add package GraphQL.MemoryCache
+```
+
+For more information see [Document Caching](https://graphql-dotnet.github.io/docs/guides/document-caching).
+
+#### 4. DataLoader
+
+DataLoader is a generic utility to be used as part of your application's data fetching layer
+to provide a simplified and consistent API over various remote data sources such as databases
+or web services via batching and caching.
+
+```
+> dotnet add package GraphQL.DataLoader
+```
+
+For more information see [DataLoader](https://graphql-dotnet.github.io/docs/guides/dataloader).
+
+> *Note: Prior to version 4, the contents of this package was part of the main GraphQL.NET package.*
```

Comment:
```suggestion
> *Note: Prior to version 4, the contents of this package were part of the main GraphQL.NET package.*
```

@Shane32 Was or were?

Reply:
"were" sounds better to me, but I'm not positive.